src/nxt_sprintf.c

```diff
@@ -152,10 +153,12 @@ nxt_vsprintf(u_char *buf, u_char *end, const char *fmt, va_list args)
         case 's':
             p = va_arg(args, u_char *);
 
-            if (nxt_fast_path(p != NULL)) {
-                while (*p != '\0' && buf < end) {
-                    *buf++ = *p++;
-                }
+            if (nxt_slow_path(p == NULL)) {
+                goto copy;
+            }
+
+            while (*p != '\0' && buf < end) {
+                *buf++ = *p++;
             }
 
             fmt++;
```

Comment:
This should be run in case of `(p == NULL)` too, so I suggest:

```c
case 's':
    fmt++;
    p = va_arg(args, u_char *);

    if (nxt_slow_path(p == NULL) {
        goto copy;
    }

    while (*p != '\0' && buf < end) {
        *buf++ = *p++;
    }
    continue;

Reply:
Yes, you are right, fixed.